### PR TITLE
HomeWork5.Fitness tracker

### PR DESCRIPTION
**Pull request by InsanireDevices**
изображение

1. Collections. LINQ.
2. 20.01.2021 -> 21.01.2021
3. Collections used to store information about User exercises. Sorting exercises in training with LINQ. But still not sure that I use that all in the right cases.

### CODE REVIEW

File: src/TMS.DotNet06.FitnessTracker/Core/Services/StatisticService.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        /// Method to get avarage statistic about User in exercises
+        /// </summary>
+        /// <param name="user">User whose avarage statistic you whant to get</param>
+        /// <returns></returns>
+        public double GetAverageStatistic(User user, Exercises exercise)
+        {
+            switch(exercise)
+            {
+                case Exercises.PushUps:
+                    double pushUpsCount = user.AvaragePushUps;
```
Comment (diman84) [inline src/TMS.DotNet06.FitnessTracker/Core/Services/StatisticService.cs:31]: Please explain how it should work, you use AvaragePushUps (and similar) property inside the method that is basically calculates it when AddNewExercise() is called

File: src/TMS.DotNet06.FitnessTracker/Core/Services/StatisticService.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                    foreach (var item in user.PushUps)
+                    {
+                        pushUpsCount += item.Units;
+                    }
+                    StatusNotification?.Invoke("Push-ups statistic updated!");
+                    return pushUpsCount / user.PushUps.Count;
```
Comment (diman84) [inline src/TMS.DotNet06.FitnessTracker/Core/Services/StatisticService.cs:37]: this is basically Average() method result

**Review by diman84 — COMMENTED** [review]
Please rethink the usage of generics to define Exercise class, it should simplify working with classes but you are violating this generic behavior all over the code (casting units to (int) for example or switching by type of exercise instead of using generic properties of base class) adding more complexity to the design

File: src/TMS.DotNet06.FitnessTracker/Core/Services/StatisticService.cs (hunk 2 above)
Comment (InsanireDevices) [reply src/TMS.DotNet06.FitnessTracker/Core/Services/StatisticService.cs:37]: Thank you, got this.

File: src/TMS.DotNet06.FitnessTracker/Core/Services/StatisticService.cs (hunk 1 above)
Comment (InsanireDevices) [reply src/TMS.DotNet06.FitnessTracker/Core/Services/StatisticService.cs:31]: Tnx, understood logical error.

**Pull request merged**